pypeit/core/flexure.py

```diff
@@ -303,147 +305,226 @@ def spec_flex_shift(obj_skyspec, arx_skyspec, arx_lines, mxshft=20):
-                flex_dict[key].append(fdict[key])
-            flex_dict['sky_spec'].append(new_sky)
-
+                    sv_fdict = copy.deepcopy(fdict)
+
+                # Punt?
+                if punt:
+                    break
+
+                # Interpolate
+                new_sky = sobj.apply_spectral_flexure(fdict)
+                if slitShift is not None:
+                    sobj.update_flex_shift(slitShift[islit])
+                # Update dict
+                for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
+                    flex_dict[key].append(fdict[key])
+                flex_dict['sky_spec'].append(new_sky)
+
+        # Check if we need to go back
+        if not slit_cen:
+            # Do we need to go back?
+            for items in return_later_sobjs:
+                if sv_fdict is None:
+                    msgs.info("No flexure corrections could be made")
+                    break
+                # Setup
+                msgs.error("This probably needs to be updated")
+                slit, ss = items
+                flex_dict = flex_list[slit]
+                sobj = specobjs[ss]
+                # Copy me
+                fdict = copy.deepcopy(sv_fdict)
+                # Interpolate
+                new_sky = sobj.apply_spectral_flexure(fdict)
+                if slitShift is not None:
+                    sobj.update_flex_shift(slitShift[islit])
+                # Update dict
+                for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
+                    flex_dict[key].append(fdict[key])
+                flex_dict['sky_spec'].append(new_sky)
+
+        # Append, this will be an empty dictionary if the flexure failed
         flex_list.append(flex_dict.copy())
 
-        # Do we need to go back?
-        for items in return_later_sobjs:
-            if sv_fdict is None:
-                msgs.info("No flexure corrections could be made")
-                break
-            # Setup
-            msgs.error("This probably needs to be updated")
-            slit, ss = items
-            flex_dict = flex_list[slit]
-            specobj = specobjs[ss]
-            sky_wave = specobj.BOX_WAVE #.to('AA').value
-            # Copy me
-            fdict = copy.deepcopy(sv_fdict)
-            # Interpolate
-            new_sky = specobj.flexure_interp(sky_wave, fdict)
-            # Update dict
-            for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
-                flex_dict[key].append(fdict[key])
-            flex_dict['sky_spec'].append(new_sky)
-
-    return flex_list
-
-
-def spec_flexure_qa(specobjs, slitords, bpm, basename, det, flex_list,
-               slit_cen=False, out_dir=None):
+    return waveimg_flex_corr, flex_list
+
+
+def spec_flexure_corrQA(ax, this_flex_dict, cntr, name):
+    # Fit
```

Comment:
Can we indicate via the QA somehow which method was used. In other words, how will the QA look if slit vs object methods are used?